### ISSUE TITLE
Bugs and questions?

### ISSUE THREAD

**Issue by erikdubbelboer** [myHelloGold/Foundation#1]
In `createTokens` the full refund is subtracted from `ethRaised` while only the value (without the refund) [was added](https://github.com/myHelloGold/Foundation/blob/a3264c75c51606cd531dfc643193d36a1402a3e7/HelloGoldSale.sol#L179-L186) in the first place. This allows me to set `ethRaised` to 0 by sending `ethRaised + 10 eth` to the contract. See:
https://github.com/myHelloGold/Foundation/blob/a3264c75c51606cd531dfc643193d36a1402a3e7/HelloGoldSale.sol#L220

----

Is this comment incorrect or the code incorrect?
https://github.com/myHelloGold/Foundation/blob/a3264c75c51606cd531dfc643193d36a1402a3e7/HelloGoldSale.sol#L128
`MaxCoinsR1` and `coinsRemaining` are both set to `80 * 10**6 * 10**8` but `minimumCap = 0` so `didOK` will always be `true`.

----

And one last question. Why can I only only `approve` 0 or when it is 0 already `approve` some other amount. Why not allow to `approve` any value?
https://github.com/myHelloGold/Foundation/blob/a3264c75c51606cd531dfc643193d36a1402a3e7/GoldBackedToken.sol#L241-L242

**Comment by DaveAppleton** [comment]
1) - yeah. I know. It was only for display but was a bugger because we had to do a workaround once we realised. It slipped past us and the auditors.
2) The comment. We had already sold out in the pre-sale so rather than changing the code (dangerous at the last minute) we just set min cap to zero. Never trust my comments.
3) This addresses a known vulnerability that if I have approved somebody and they have spent some of that approval, changing to a non zero value effectively re-writes their allowance allowing them to spend extra. It is a real edge case but it could happen.
